### ISSUE TITLE
params.auth0, cookies().getAll(), and  cookies().set('appSession', ...)`. `cookies() should be awaited errors - Next 15.0.2

### ISSUE THREAD

**Issue by sterno2510** [auth0/nextjs-auth0#1802]
### Checklist

- [X] The issue can be reproduced in the [nextjs-auth0 sample app](https://github.com/auth0-samples/auth0-nextjs-samples/tree/main/Sample-01) (or N/A).
- [X] I have looked into the [Readme](https://github.com/auth0/nextjs-auth0#readme), [Examples](https://github.com/auth0/nextjs-auth0/blob/main/EXAMPLES.md), and [FAQ](https://github.com/auth0/nextjs-auth0/blob/main/FAQ.md) and have not found a suitable solution or answer.
- [X] I have looked into the [API documentation](https://auth0.github.io/nextjs-auth0/) and have not found a suitable solution or answer.
- [X] I have searched the [issues](https://github.com/auth0/nextjs-auth0/issues) and have not found a suitable solution or answer.
- [X] I have searched the [Auth0 Community](https://community.auth0.com) forums and have not found a suitable solution or answer.
- [X] I agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).

### Description

When using the getSession function imported from auth0/nextjs-auth0, I get a few errors in the console when the app compiles.  

Specifically these 3:
1. Error: Route "/dashboard" used `cookies().getAll()`. `cookies()` should be awaited before using its value. Learn more: https://nextjs.org/docs/messages/sync-dynamic-apis
2.  Error: Route "/dashboard" used `cookies().set('appSession', ...)`. `cookies()` should be awaited before using its value. Learn more: https://nextjs.org/docs/messages/sync-dynamic-apis
3. Error: Route "/api/auth/[auth0]" used `params.auth0`. `params` should be awaited before using its properties. Learn more: https://nextjs.org/docs/messages/sync-dynamic-apis

All three errors appear to be coming from here:

at get (file:///Users/brianstern/code/ai-companion-next/node_modules/next/src/server/base-server.ts:3121:48)
```
  29 |     ...args: A
  30 |   ): ReturnType<F> {
> 31 |     return fn.call(thisArg, ...args);
     |              ^
  32 |   }
  33 |
  34 |   bind<T>(_context: types.Context, target: T): T {
```

### Reproduction

1. Use getSession()
2. Run app with npm run dev
3. Go to route and errors appear in console.

### nextjs-auth0 version

3.5.0

### Next.js version

15.0.2

### Node.js version

22.8.0

**Comment by nicolopadovan** [comment]
The same issue arises when using any of the other functions that use `headers()` or `cookies()`, such as `getAccessToken` and similar.
This is due to the [breaking changes](https://nextjs.org/blog/next-15#async-request-apis-breaking-change) that were implemented in NextJS 15.

**Comment by MrSimmmons** [comment]
Many discussions have been made over at https://github.com/auth0/nextjs-auth0/issues/1779 

TLDR; NextJS 15 _**IS NOT SUPPORTED**_ until the v4 of the SDK is released.

V4 is effectively a rewrite and has many breaking changes, but is in beta. So id recommend using the beta if you want to use Next 15, or waiting for the official release whenever that happens

**Comment by brth31** [comment]
Thanks for reporting this issue. These errors are related to Next.js 15's new requirement for async handling of cookies and headers, which is a breaking change. As noted in our [announcement](https://github.com/auth0/nextjs-auth0/issues/1808), Next.js 15 compatibility is only available in our v4 beta release. 

Closing this issue as it's being tracked as part of the broader Next.js 15 compatibility work. For v4 beta specific issues, please open new issues and tag them accordingly.
